examples/sub-class.py

```diff
@@ -29,36 +29,30 @@
         sys.path.insert(0, cmd_subfolder)
     import paho.mqtt.client as mqtt
 
-class MyMQTTClass:
-    def __init__(self, clientid=None):
-        self._mqttc = mqtt.Client(clientid)
-        self._mqttc.on_message = self.mqtt_on_message
-        self._mqttc.on_connect = self.mqtt_on_connect
-        self._mqttc.on_publish = self.mqtt_on_publish
-        self._mqttc.on_subscribe = self.mqtt_on_subscribe
+class MyMQTTClass(mqtt.Client):
 
-    def mqtt_on_connect(self, mqttc, obj, flags, rc):
+    def on_connect(self, mqttc, obj, flags, rc):
```

Comment:
Right, I see. You end up with `on_connect(self, self, ...)` effectively. That's pretty ugly. Is there any way to make it so that the inheritance-based interface doesn't need the repeated argument.

Might it be better to create an intermediate method, say `self.handle_connect(*args)`, as a regular method, that has a default implementation that calls `self.on_connect(self, *args)`.